tests/OpportunityContextUtils.test.ts

```diff
@@ -223,4 +223,174 @@ describe('OpportunityContextUtils', () => {
       });
     });
   });
+
+  describe('external provider properties', () => {
+    describe('initOpportunityContext', () => {
+      it('external created at context will be set', () => {
+        initOpportunityContext(ctx, {
+          key: OpportunityContextKeys.EXTERNAL_CREATED_AT,
+          value: new Date('2021-01-01'),
+        });
+        expect(ctx.externalCreatedAt).toBe('some-value');
```

Comment:
You will need to use `jest.useFakeTimers('modern')` to test dates confidently.

Also `new Date('2021-01-01').toString() !== 'some-value'` so this test will fail.